### ISSUE TITLE
Fix false positives for CSS in HTML in *-empty-line-before

### ISSUE THREAD

**Issue by gucong3000** [stylelint/stylelint#2853]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`*-empty-line-before` not support for HTML

> Which rule, if any, is this issue related to?

`rule-empty-line-before`, `at-rule-empty-line-before`

> What CSS is needed to reproduce this issue?

It is CSS in HTML.

```html
<style>
h1 {
  font-weight: normal;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

```js
module.exports = {
  extends: 'stylelint-config-standard',
  // add your custom rules here
  rules: {
  }
};

```

> Which version of stylelint are you using?

`8.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

```shell
stylelint "**/*.{vue,html,css,sss,less,scss,sass}" --custom-syntax postcss-html
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no.

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following errors were flagged:"
https://circleci.com/gh/gucong3000/webpack/16#artifacts/containers/0

```shell
src/App.vue
 15:1  ✖  Expected empty line before rule   rule-empty-line-before

src/components/Hello.vue
 36:1  ✖  Expected empty line before rule   rule-empty-line-before
```
